packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -772,6 +772,23 @@ CodeGuru profiling is supported for all Java runtimes and Python3.6+ runtimes.
 See [the AWS documentation](https://docs.aws.amazon.com/codeguru/latest/profiler-ug/setting-up-lambda.html)
 to learn more about AWS Lambda's Profiling support.
 
+## Lambda with SnapStart
+
+SnapStart is currently supported only on Java 11 runtime. After you enable Lambda SnapStart for a particular Lambda function, publishing a new version of the function will trigger an optimization process.
+
+See [the AWS documentation](https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html) to learn more about AWS Lambda SnapStart
+
+```ts
+const fn = new lambda.Function(this, 'MyFunction', {
+  code: lambda.Code.fromAsset('handler.zip'),
+  runtime: lambda.Runtime.JAVA_11,
+  handler: 'example.Handler::handleRequest',
+  snapStart: true,
+});
+
+const version = fn.currentVersion;
```

Comment:
Nice! Good to know fn.currentVersion will actually create a new version if no existing version.

Reply:
This is actually another great reason for this to not be a prop. For this to be a prop, you should be able to create a brand new Lambda function with this prop enabled, and the deployment should succeed.

This won't work though, because to enable start, a version has to already be published, and that needs an existing function. Normally, CFN can create the lambda function and the version at once since it can figure out the dependency, but this is a circular dependency; the Version depends on the Function, but the Function also depends on the Version because snap start is a property on the function itself. 

(unless of course CFN figures this out for us, which I will check manually later, but it's not clear right now). 

So, in summary: we need a method `enableSnapStart()` that does this error checking and clearly documents the restrictions, including the restriction that a lambda function must already have been deployed to enable this (assuming this restriction actually exists, again I'll verify this later).